src/modules/SpaceCalculation.rb

```diff
@@ -578,183 +559,147 @@ def get_partition_info
-                  free_size -= used
-
-                  # check for underflow
-                  if free_size < 0
-                    log.info "Fixing free size: #{free_size} to 0"
-                    free_size = 0
-                  end
-                end
+        used = 0
+        # If reusing a previously existent filesystem
+        if filesystem.exists_in_probed?
 
-                # convert into KiB for TargetInitDU
-                free_size_kib = free_size / 1024
-                used_kib = used / 1024
-                mount_name = part["mount"]
-                log.info "partition: mount: #{mount_name}, free: #{free_size_kib}KiB, used: #{used_kib}KiB"
+          # Mount the filesystem to check the available space.
 
-                mount_name = mount_name[1..-1] if remove_slash && mount_name != "/"
+          tmpdir = SCR.Read(path(".target.tmpdir")) + "/diskspace_mount"
+          SCR.Execute(path(".target.bash"), "mkdir -p #{Shellwords.escape(tmpdir)}")
 
-                target_partitions << {
-                  "filesystem" => used_fs.to_s,
-                  "growonly" => growonly,
-                  "name" => mount_name,
-                  "used" => used_kib,
-                  "free" => free_size_kib
-                }
+          # mount options determined by partitioner
+          mount_options = filesystem.fstab_options.to_a
+
+          # mount in read-only mode (safer)
+          mount_options << "ro"
+
+          # add "nolock" if it's a NFS share (bnc#433893)
+          if used_fs == :nfs
+            log.info "Mounting NFS with 'nolock' option"
+            mount_options << "nolock"
+          end
+
+          # join the options
+          mount_options_str = mount_options.uniq.join(",")
+
+          # Use DM device if it's encrypted, plain device otherwise
+          # (bnc#889334)
+          #device = part["crypt_device"] || part["device"] || ""
+          device = filesystem_dev_name(filesystem)
+
+          mount_command = "mount -o #{mount_options_str} " \
+            "#{Shellwords.escape(device)} #{Shellwords.escape(tmpdir)}"
+
+          log.info "Executing mount command: #{mount_command}"
+
+          result = SCR.Execute(path(".target.bash"), mount_command)
+          log.info "Mount result: #{result}"
+
+          if result == 0
+            # specific handler for btrfs
+            if used_fs == :btrfs
+              used = btrfs_used_size(tmpdir)
+              free_size -= used
+              growonly = btrfs_snapshots?(tmpdir)
+            else
+              partition = SCR.Read(path(".run.df"))
+
+              Builtins.foreach(partition) do |p|
+                if p["name"] == tmpdir
+                  log.info "Partition: #{p}"
+                  free_size = p["free"].to_i * 1024
+                  used = p["used"].to_i * 1024
+                end
               end
             end
+
+            SCR.Execute(path(".target.bash"), "umount #{Shellwords.escape(tmpdir)}")
```

Comment:
Same again. Old code, I don't think now is the right moment to change it (that would make harder to sync with master in the future).